src/platform/tests/TestMdns.cpp

```diff
@@ -0,0 +1,81 @@
+#include <assert.h>
+#include <avahi-common/simple-watch.h>
+
+#include <avahi-client/client.h>
+#include <avahi-client/lookup.h>
+#include <avahi-common/error.h>
+#include <avahi-common/malloc.h>
+#include <avahi-common/simple-watch.h>
+
+#include "platform/CHIPDeviceLayer.h"
+#include "platform/Mdns.h"
+
+using chip::DeviceLayer::MdnsResolveResult;
+using chip::DeviceLayer::MdnsService;
+using chip::DeviceLayer::MdnsServiceProtocol;
+using chip::DeviceLayer::TextEntry;
+
+static void HandleResolve(void * context, MdnsResolveResult * result, CHIP_ERROR error)
+{
+    char addrBuf[100];
+
+    assert(result != nullptr);
```

Comment:
Could we convert these tests to NL unit tests?

I am thinking that assert would be cleaned out in release builds and I may want to run tests on release.
Is there a timeout or equivalent?I imagine we would like this to fail if we do not manage to publish and resolve properly.